### ISSUE TITLE
Ability to change per app notification settings

### ISSUE THREAD

**Issue by MrQvest** [linuxmint/cinnamon#11525]
### Is your feature request related to a problem? Please describe.
We currently have the ability to set notifications on or off globally
### Describe the solution you'd like
Would be better if one could set notifications on or off on a per application basis as well as globally.
On notification settings, there could be a setting saying "change per app notifications" that would open up a menu for selecting which applications to turn notifications on or off

**Comment by Belaial** [comment]
This is exactly something I have been missing in Cinnamon, hopefully it could be implemented someday.

Also if the notifications could be grouped per application that could also help, for example now when Teams "only" is a Progressive Web App on Linux, it would be great if Teams PWA's notifications could be displayed as a separate group / icon. This would be handy to quickly and clearly see in what applications you have unread notifications.

**Comment by alextpedro** [comment]
To add to the per-app notifications, it would be great to be able to set whether an app's notifications play a sound. For example, I want to see discord notifications but I don't want to hear the cinnamon notification sound for each one. However, if another app sends notifications, I do want to hear sound for those ones.

**Comment by Belaial** [comment]
> To add to the per-app notifications, it would be great to be able to set whether an app's notifications play a sound. For example, I want to see discord notifications but I don't want to hear the cinnamon notification sound for each one. However, if another app sends notifications, I do want to hear sound for those ones.

Agree, this would be a very hand feature to have.

**Comment by socketbox** [comment]
Someone in the Mint forum responded to a similar request by suggesting the use of the _dconf Editor_ application. However, I couldn't find anything in the way of a notifications setting for the app whose behavior I wanted to modify (_backintime_) in said editor.

**Comment by nicknugget** [comment]
That indeed would be a great feature. As of the current status quo, notifications can become quite annoying and I wish I could configure what I'm really interested in. For example I'm not interested to get informed that my Bluetooth-Mouse went into sleep and disconnected. Or that I have a new mail. Because another applet already shows that.

**Comment by Ccjr0** [comment]
The printer detection notification is particularly guilty of this. Most other DEs don't have the issue in the same way, but maybe if you could at least put a 'don't show this again' button on the notification window similar to the 'connection established' notif that might work too.

**Comment by Secret-chest** [comment]
Also, there should be an easy way to show them again

**Comment by Nebula-0-System** [comment]
came here to request exactly this. would love to see this implemented. even if it were a manual filter like matching strings. gnome offers central control and it's honestly the only reason i'd use gnome over Cinnamon.

**Comment by Arcaravan** [comment]
> Esto es precisamente algo que he echado de menos en Cinnamon; ojalá se pueda implementar algún día.
> 
> Además, si las notificaciones se pudieran agrupar por aplicación, sería muy útil. Por ejemplo, ahora que Teams es solo una aplicación web progresiva en Linux, sería ideal que las notificaciones de Teams PWA se mostraran como un grupo o icono aparte. Esto facilitaría ver de forma rápida y clara en qué aplicaciones hay notificaciones sin leer.

Pués a 8 de Noviembre de 2025 aún siguen sin implementarlo.

**Comment by Arcaravan** [comment]
Asof today, 8 de november de 2025, it has still not been implemented,

**Comment by Arcaravan** [comment]
On Spotify , everytime a new song starts.

**Comment by hfvmarques** [comment]
> On Spotify , everytime a new song starts.

Oh my gods yes!

**Comment by dktapps** [comment]
> On Spotify , everytime a new song starts.

Found my way here because of this, on my travels I discovered there is a toggle - if you click your avatar -> Settings -> about halfway down there's a Display section -> Show desktop notifications when the song changes

I will say I'm loving Mint so far, but this feature request is about the biggest thing I'm missing from Windows

**Comment by hulumibuna** [comment]
Every time my BT headset disconnects/reconnects, I get 2 notifications (and this happens dozens of times a day).
I've been able to get rid of half of the notifications already by disabling the ConnectionNotifier in blueman-applet, but when I disconnect the bluetooth adapter, I'm still getting notifications when I restart the bluetooth adapter (sometimes I need to use this way to allow my WH-1000XM4 to connect to my other computer).
I want notifications for my calendar items and so on, things that need my attention. When the notifications get overwhelmed by a headset that disconnects all the time, I might as well just disable notifications altogether.
In its current state, notifications are useless to me unless I can define what I don't want notifications for.

**Comment by ItsZyad** [comment]
Bump

**Comment by Ljuk67** [comment]
Yep, this would be handy to have. I want to see system notifications, but not "new email" notifications for example. The email app doesn't provide that and I feel it should be possible to disable notifications per app on the system level.

**Comment by LukasMino** [comment]
Found this today and am sad that this is not implemented 3 years later. It is such a basic QoL feature for me and might be one of the reasons I actually switch to a different desktop. 

In addition to what others already said, here are some other helpful features for this system:
* Grouping notifications, e.g. by application. If I have 20+ notifications just for Bluetooth devices (dis)connecting and printing, I likely miss the one or two actually important notifications I get.
* Make application specific notifications very customizable, make us be able to choose if an applications notification ... 
  * plays a sound (also depending on "do not disturb" or fullscreen),
  * is displayed on the screen (also depending on "do not disturb" or fullscreen),
  * is saved in the notification applet or just vanishes after a given time
  * is low, normal or high priority
* In addition to that, having a notification history would be nice (not like the applet)

As there are already good notification system on other desktops (be it on other Linux Desktops or even Windows and Mac) there is plenty of inspiration out there as well.

**Comment by fredcw** [comment]
I'm working on per app notification settings atm, PR coming soon.

**Comment by ashalexemail** [comment]
@fredcw 
Any update on this?

**Comment by fredcw** [comment]
Fixed with https://github.com/linuxmint/cinnamon/pull/13598

**Comment by Ljuk67** [comment]
Great, thanks! What does that mean for us end users? I'm on 22.3, ran the update manager, installed every update but don't see any chnage.

And the referenced pull is hard to read for outsiders, no instructions on what to do to get the new app-notification menu.

**Comment by fredcw** [comment]
@Ljuk67 It will be in cinnamon 6.8 (mint 23)

**Comment by ashalexemail** [comment]
@fredcw Great, will there possibly be a 22.4 version, since mint 23 is a while away?
Any thoughts

Any did you make this all by yourself?

If so, congrats :D
